### PR TITLE
[#74] feat: add imported css to its importer

### PR DESCRIPTION
**Pull request by mseyfayi**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -103,30 +86,44 @@ export default function cssInjectedByJsPlugin({
...
             }
 
-            jsAssetTargets.forEach((jsAsset) => {
+            for (const jsAsset of jsAssetTargets) {
+                // @ts-ignore
+                const cssNamesSet: Set<string> = jsAsset.viteMetadata.importedCss;
+                let cssToInject: string = '';
+                cssNamesSet.forEach((cssName) => {
+                    cssToInject += String(extractCssCode(bundle[cssName] as OutputAsset));
+                });
```
Comment (marco-prontera) [inline src/index.ts:101]: I think you can delete directly here the OutputAsset once the code is extracted

File: src/index.ts [hunk 2]
```diff
@@ -103,30 +86,44 @@ export default function cssInjectedByJsPlugin({
...
+            }
+            cssAssets.forEach((css) => {
+                delete bundle[css];
             });
         },
     };
 }
 
+const extractCssCode = (cssAsset: OutputAsset) =>
+    typeof cssAsset.source == 'string' ? cssAsset.source.replace(/(\r\n|\n|\r)+$/gm, '') : cssAsset.source;
```
Comment (marco-prontera) [inline src/index.ts:125]: Use function declaration with name instead of an anonymous function

**Review by marco-prontera — CHANGES_REQUESTED** [review]
First of all, thank you for your contribution! 👍

We need to make this behavior configurable since currently the library is used with the existing behavior. 

What we can do is:
- Make this behavior configurable (I know you think this is a bug, but we need to think of this as an improvement)

Someone could have the need to inject the CSS once and for all since this can impact performance in certain cases. The configuration flag can be named as 'relativeCSSInjection' a boolean when true the behavior you proposed will be enabled.

File: src/index.ts (hunk 1 above)
Comment (mseyfayi) [reply src/index.ts:101]: Some CSS bundles can be used in several JS assets, if we delete the bundle here other JS assets can't use the CSS.  
For example, I have a `Button` that uses `Spinner` for loading, the button should use both `spinner.css` and `button.css` and the plugin will remove `spinner.css` from bundles and while vite is bundling the spinner, `spinner.css` is removed.

Comment (marco-prontera) [reply src/index.ts:101]: You're right, sorry I didn't think about it

**Comment by mseyfayi** [comment]
> First of all, thank you for your contribution! 👍
> 
> We need to make this behavior configurable since currently the library is used with the existing behavior.
> 
> What we can do is:
> 
> * Make this behavior configurable (I know you think this is a bug, but we need to think of this as an improvement)
> 
> Someone could have the need to inject the CSS once and for all since this can impact performance in certain cases. The configuration flag can be named as 'relativeCSSInjection' a boolean when true the behavior you proposed will be enabled.


`relativeCSSInjection` has been added, and it will switch between adding all CSSs or just imported CSSs

**Review by marco-prontera — APPROVED** [review]
Perfect!
Give me the time to test locally and write documentation and I will notify you of the new release!

**Comment by marco-prontera** [comment]
@mseyfayi Does this feature work with clean usage (without vue or anything) or am I doing something wrong? If you have time can you give me an example? A test repository because with my tests I can't get the desired result, maybe I'm doing something wrong.

Thanks in advance!

**Comment by mseyfayi** [comment]
I apologize for not getting back to you sooner.  
I only test it in React, and you're right it should be tested in Vue and Svelte, and other frameworks.  But since we are working with the Rollup, not the framework plugins, I don't think there would be any problem.

But still, I'm not sure!

**Pull request merged**